src/core/Akka.Cluster.Tests/ClusterLogSpec.cs

```diff
@@ -68,7 +68,7 @@ protected async Task JoinAsync(string expected)
             {
                 await EventFilter
                     .Info(contains: expected)
-                    .ExpectOneAsync(async () => _cluster.Join(_selfAddress));
+                    .ExpectOneAsync(async () => await _cluster.JoinAsync(_selfAddress).ShouldCompleteWithin(Remaining));
```

Comment:
I have little trust that the `JoinAsync` method is actually correct. I think we should use a more manual way of asserting this, like the one you've implemented in the `DownAsync` method below.